File: Python/2021/9thOct_Shrivis.py
def GCD(a, b):
	return a if b==0 else GCD(b, a%b)		

def fact(n):
    return 1 if (n == 1 or n == 0) else n * fact(n - 1);

# input format: a b
a, b = map(int, input().split())
print(GCD(fact(a), fact(b)))

File: C++/2021/9thOct_Shrivis.cpp
#include <iostream>

using namespace std;

int fact(int n) {
    if (0 <= n && n < 2) return 1;
    return n*fact(n-1);
}

int GCD(int a, int b) {
    if (b == 0) return a;
    return GCD(b, a%b);
}

int main() {
    int a, b;
    cin>>a>>b;
    cout<<GCD(fact(a), fact(b));
    return 0;
}
